### ISSUE TITLE
Fix circular dependency warning

### ISSUE THREAD

**Issue by yhara** [biwascheme/biwascheme#211]
```
$ make -B
...
(!) Circular dependencies
src/header.js -> src/system/set.js -> src/system/error.js -> src/system/_writer.js -> src/header.js
src/header.js -> src/system/set.js -> src/system/error.js -> src/system/_writer.js -> src/system/_types.js -> src/header.js
src/system/_writer.js -> src/system/_types.js -> src/system/_writer.js
...and 11 more
...
```

**Comment by yhara** [comment]
> src/header.js -> src/system/set.js -> src/system/error.js -> src/system/_writer.js -> src/header.js

- header.js needs set.js to implement `nil.to_set`
- set.js needs error.js to raise error
- error.js needs `to_write` of _writer.js to format an object for error message
- _writer.js needs header.js for `nil` and `undef`

**Comment by yhara** [comment]
Most warnings are fixed by #227. There is one more warning left, but I'm closing this ticket because the last one

```
src/system/_types.js -> src/system/_writer.js -> src/system/_types.js
```

needs #228 to be fixed.
